test/test_signal.py

```diff
@@ -0,0 +1,45 @@
+import asyncio
+import awaitwhat
+
+
+FUTURES = []
+
+
+async def main():
+    await do_work()
+
+
+async def do_work():
+    futs = [frob_a_tree() for i in range(1)]
+    await asyncio.gather(*futs)
+
+
+async def frob_a_tree():
+    await b_tree()
+
+
+async def b_tree():
+    f = asyncio.Future()
+    FUTURES.append(f)
+    await asyncio.shield(frob_a_branch(f))
```

Comment:
Can the test be simplified?
Perhaps it's enough to have `Task→coro→Future` as async code, graph of which is being dumped?